util/src/main/java/org/folio/metastorage/util/OaiParser.java

```diff
@@ -0,0 +1,142 @@
+package org.folio.metastorage.util;
+
+import java.io.InputStream;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+public class OaiParser<T> {
```

Comment:
@adamdickmeiss I wonder if it makes sense to refactor this now and put the parsers into individual classes: ListRecordsParser, ListIdentifiersParser, .,, etc. Possibly with a base class to handle common protocol elements. In case we need to implement additional operations like Identify or ListMetadataFormats.

Also, might be a good idea to move Oai stuff into it's own package.

Reply:
I'll do this very soon. I don't think sub classing is really useful here. But we need more get-methods to dig out the response.